product_configurator/models/product_config.py

```diff
@@ -582,10 +580,36 @@ def update_session_configuration_value(self, vals, product_tmpl_id=None):
                 custom_val_dict.update({attr_id: val})
                 # Ensure there is no standard value stored if we have switched
                 # from selected value to custom value.
-                attr_val_dict.update({attr_id: False})
+                attr_val_dict.update({attr_id: custom_val.id})
 
         self.update_config(attr_val_dict, custom_val_dict)
 
+    def _update_field_values(self, vals, field_name, attr_line):
+        """New method for update field values for a given attribute."""
+        final_val = None
+        if not vals[field_name]:
+            return final_val
+        # Retrieve existing values for the attribute
+        value_ids = self.value_ids.filtered(
+            lambda value, attr_line=attr_line: value.attribute_id.id
+            == attr_line.attribute_id.id
+        )
+        # Initialize `final_val` with IDs
+        final_val = value_ids.ids or []
+
+        # Process each `field_vals` operation for the current attribute
+        for field_vals in vals.get(field_name, []):
+            if field_vals and field_vals[0] == Command.SET:
+                final_val = list(set(final_val + (field_vals[2] or [])))
+            elif field_vals and field_vals[0] == Command.LINK:
+                if field_vals[1] not in final_val:
+                    final_val.append(field_vals[1])
+            elif field_vals and field_vals[0] == Command.UNLINK:
```

Comment:
Don't we need to check for both?

```suggestion
            elif field_vals and field_vals[0] in (Command.UNLINK, Command.DELETE):
```

See: https://github.com/odoo/odoo/blob/21033abc8499ac846e567d981b30b39fe92c87ff/addons/web/models/models.py#L1232

Reply:
Basically, we have M2m fields and not One2many as dynamic. Still, for safety, we have also added a delete command.